packages/contracts/test/manager/facets/ManagerFacet.t.sol

```diff
@@ -0,0 +1,203 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../DiamondTestSetup.sol";
+import "../../../src/dollar/interfaces/ICurveFactory.sol";
+import "../../../src/dollar/interfaces/IMetaPool.sol";
+import "../../../src/dollar/mocks/MockDollarToken.sol";
+import "../../../src/dollar/mocks/MockTWAPOracleDollar3pool.sol";
+
+import {GOVERNANCE_TOKEN_MINTER_ROLE, GOVERNANCE_TOKEN_BURNER_ROLE} from "../../../src/manager/libraries/LibAppStorage.sol";
+
+contract TestManagerFacet is DiamondSetup {
+    function testCanCallGeneralFunctions() public {
+        IManagerFacet.getExcessDollarsDistributor(contract1);
+    }
+
+    function testShouldSetTwapOracleAddress() public prankAs(admin) {
+        MockDollarToken dollarToken;
+        dollarToken = new MockDollarToken(10000);
+        // deploy twapPrice oracle
+        MockTWAPOracleDollar3pool _twapOracle = new MockTWAPOracleDollar3pool(
+            address(0x100),
+            address(dollarToken),
+            address(0x101),
+            100,
+            100
+        );
+        IManagerFacet.setTwapOracleAddress(address(_twapOracle));
+        assertEq(IManagerFacet.getTwapOracleAddress(), address(_twapOracle));
+    }
+
+    function testShouldSetDollarTokenAddress() public prankAs(admin) {
```

Comment:
can we have a test that assert we can't set if not admin ?

Reply:
That would be testing the AccessControl contracts which is outside of the scope. If we're using untrusted libraries then those should be audited and tested before being implemented anywhere.

Reply:
> can we have a test that assert we can't set if not admin ?

Already tested it when writing test cases, but didn't add it. Ofc, it can be added